files/setup-02-proxy.sh

```diff
@@ -23,28 +23,42 @@ if [ -n "${HTTP_PROXY}" ] || [ -n "${HTTPS_PROXY}" ]; then
     fi
 
     if [ ! -z "${HTTP_PROXY}" ]; then
-        if [ $YES_CREDS -eq 1 ]; then
-            HTTP_PROXY_URL="http://${PROXY_USERNAME}:${PROXY_PASSWORD}@${HTTP_PROXY}"
-        else
-            HTTP_PROXY_URL="http://${HTTP_PROXY}"
-        fi
-        echo "HTTP_PROXY=\"${HTTP_PROXY_URL}\"" >> ${PROXY_CONF}
-        cat > ${DOCKER_PROXY}/http-proxy.conf << __HTTP_DOCKER_PROXY__
+        a=(`printf '%s\n' "${HTTP_PROXY//\:\/\//$'\n'}"`)
+        if [ ${#a[*]} -eq 2 ]; then
+            HTTP_PROXY_PREFIX=${a[0]}
+            HTTP_PROXY=${a[1]}
```

Comment:
For ease of debugging, would you mind creating a brand new variable like `HTTP_PROXY_SERVER_PORT` rather than reusing `HTTP_PROXY`? I had to take a closer look at the code to see what was happening and was confused on why HTTP_PROXY was being redefined.